### PR TITLE
Added 'context' to loop annotations.

### PR DESCRIPTION
**Pull request by e-ngo**
Previously, function calls were falsely considered as loops. This PR fixes this issue by considering the 'context' of the call. If a jumpdest was reached by a different instruction, it would not add to its loop annotation.

### CODE REVIEW

File: mythril/laser/ethereum/state/machine_state.py [hunk 1]
```diff
@@ -224,6 +228,19 @@ def __str__(self):
...
+    @property
+    def pc(self) -> int:
+        """
+
+        :return:
+        """
+        return self._pc
+
+    @pc.setter
+    def pc(self, value):
```
Comment (norhh) [inline mythril/laser/ethereum/state/machine_state.py]: cool

File: mythril/laser/ethereum/strategy/extensions/bounded_loops.py [hunk 2]
```diff
@@ -64,22 +64,30 @@ def get_strategic_global_state(self) -> GlobalState:
...
-            address = state.get_current_instruction()["address"]
+            cur_instr = state.get_current_instruction()
 
-            if address in annotation._reached_count:
-                annotation._reached_count[address] += 1
+            if cur_instr["opcode"].upper() != "JUMPDEST":
+                return state
+
+            # create unique instruction identifier
+            key = "{}{}{}".format(
```
Comment (JoranHonig) [inline mythril/laser/ethereum/strategy/extensions/bounded_loops.py]:
```suggestion
            key = "{};{};{}".format(
```
Address and prev_pc are both integers, I believe that without delimiters there is a slight chanche for collision

Comment (norhh) [reply mythril/laser/ethereum/strategy/extensions/bounded_loops.py]: Oh! Yeah :rofl:

Comment (e-ngo) [reply mythril/laser/ethereum/strategy/extensions/bounded_loops.py]: Nice, thanks for the suggestion! Changed!

File: mythril/laser/ethereum/state/machine_state.py (hunk 1 above)
Comment (e-ngo) [reply mythril/laser/ethereum/state/machine_state.py]: Thanks

**Pull request merged**